service-front/module/Application/src/Helpers/LpaFormHelper.php

```diff
@@ -0,0 +1,263 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Application\Helpers;
+
+use Application\Helpers\DTO\LpaFormHelperResponseDto;
+use Application\Services\SiriusApiService;
+use Laminas\Form\FormInterface;
+use Laminas\Stdlib\Parameters;
+use Laminas\Http\Response;
+
+class LpaFormHelper
+{
+    public function findLpa(
+        string $uuid,
+        Parameters $formData,
+        FormInterface $form,
+        array $siriusCheck,
+        array $detailsData,
+    ): LpaFormHelperResponseDto {
+        $form->setData($formData);
+        $result = [];
+
+        if ($form->isValid()) {
+            if (
+                ! array_key_exists('opg.poas.lpastore', $siriusCheck) ||
+                empty($siriusCheck['opg.poas.lpastore'])
+            ) {
+                $result = [
+                    'status' => 'Not Found',
+                    'message' => "No LPA found.",
+                ];
+                return new LpaFormHelperResponseDto(
+                    $uuid,
+                    $form,
+                    [
+                        'lpa_response' => $result
+                    ],
+                );
+            }
+
+            if (! $this->checkLpaNotAdded($form->get('lpa')->getValue(), $detailsData)) {
+                $result = [
+                    'status' => 'Already added',
+                    'message' => "This LPA has already been added to this ID check.",
+                ];
+                return new LpaFormHelperResponseDto(
+                    $uuid,
+                    $form,
+                    [
+                        'lpa_response' => $result
+                    ],
+                );
+            }
+
+            $idCheck = $this->compareCpRecords($detailsData, $siriusCheck);
+            $statusCheck = $this->checkStatus($siriusCheck);
+            $channelCheck = $this->checkChannel($siriusCheck);
+
+            if ($statusCheck['error'] === true) {
+                $result['status'] = $statusCheck['status'];
+                $result['message'] = $statusCheck['message'];
+            } elseif ($idCheck['error'] === true) {
+                $result['status'] = 'no match';
```

Comment:
I don't think it's a good idea to overwrite external status data, otherwise we can't reliably display the original status of the case.